core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/TwitterDao.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     * Create an entity(Tweet) to the underlying storage
+     *
+     * @param tweet entity that to be created
+     * @return created entity
+     */
+    @Override
+    public Tweet create(Tweet tweet) {
+        URI uri;
+        uri = getPostUri(tweet);
+
+        //Execute HTTP Request
+        HttpResponse response = httpHelper.httpPost(uri);
+        return parseData(response, HTTP_OK);
+    }
+
+    /**
+     * Find an entity(Tweet) by its id
+     *
+     * @param s entity id
+     * @return Tweet entity
+     */
+    @Override
+    public Tweet findById(String s) {
+        URI uri;
+        try {
+            String uriStr = API_BASE_URI + SHOW_PATH + QUERY_SYM + "id" + EQUAL + s;
+            uri = new URI(uriStr);
+        } catch (URISyntaxException ex) {
+            throw new IllegalArgumentException("Invalid id input", ex);
+        }
+
+        //Execute HTTP Request
+        HttpResponse response = httpHelper.httpGet(uri);
+        return parseData(response, HTTP_OK);
+    }
+
+    /**
+     * Delete an entity(Tweet) by its ID
+     *
+     * @param s of the entity to be deleted
+     * @return deleted entity
+     */
+    @Override
+    public Tweet deleteById(String s) {
+        URI uri;
+        try {
+            String uriStr = API_BASE_URI + DELETE_PATH + "/" + s + ".json";
+            uri = new URI(uriStr);
+        } catch (URISyntaxException ex) {
+            throw new IllegalArgumentException("Invalid id input", ex);
+        }
+
+        //Execute HTTP Response
+        HttpResponse response = httpHelper.httpPost(uri);
+        return parseData(response, HTTP_OK);
+    }
+
+    private URI getPostUri(Tweet tweet) {
+        URI uri;
+        PercentEscaper percentEscaper = new PercentEscaper("", false);
+        try {
+            uri = new URI(API_BASE_URI + POST_PATH + QUERY_SYM + "status" + EQUAL + percentEscaper
+                    .escape(tweet.getText()) + AMPERSAND + "long" + EQUAL + tweet.getCoordinates()
+                    .getCoordinates().get(0) + AMPERSAND + "lat" + EQUAL + tweet.getCoordinates()
+                    .getCoordinates().get(1));
+            return uri;
+        } catch (URISyntaxException ex) {
+            throw new IllegalArgumentException("Invalid input", ex);
+        }
+    }
+
+
+    /**
+     * Takes the response data of the request and creates the tweet object from the body
+     *
+     * @param response
+     * @param httpOk
+     * @return
+     */
+    public Tweet parseData(HttpResponse response, int httpOk) {
+        Tweet tweet = null;
+        int status = response.getStatusLine().getStatusCode();
+        if (status != httpOk) {
+            try {
+                logger.error(EntityUtils.toString(response.getEntity()));
+            } catch (IOException ex) {
+                logger.error("ERROR: Response has no entity");
+            }
+            throw new RuntimeException("Unexpected HTTP status code:" + status);
```

Comment:
you might want to put this line into the catch block. please pass the caught exception to RuntimeException